safeeyes/tests/test_core.py

```diff
@@ -0,0 +1,584 @@
+from collections import deque
+import datetime
+import logging
+import pytest
+
+from safeeyes import core
+from safeeyes import model
+
+import threading
+from time import sleep
+
+from unittest import mock
+
+class TestSafeEyesCore:
+    @pytest.fixture(autouse=True)
+    def set_time(self, time_machine):
```

Comment:
Good catch with `time_machine` pytest plugin but it would be better to add it as dev/test dependency with pinned version